### PR TITLE
Forecasting: base/template docstring fixes, added fit_predict method

### PR DESCRIPTION
**Pull request by fkiraly**
Some small changes to the forecasting base class and template:

* updates to docstrings, clarifications
* added `fit_predict` method which does what one would "obviously" expect, fitting & forecast in one step.

### CODE REVIEW

File: sktime/forecasting/base/_base.py [hunk 1]
```diff
@@ -161,6 +158,38 @@ def predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
...
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        y_pred_int : pd.DataFrame - only if return_pred_int=True
+            Prediction intervals
+        """
+
+        if fh is None:
+            fh = y.index
```
Comment (mloning) [inline sktime/forecasting/base/_base.py:187]:
```suggestion
```

**Review by mloning — COMMENTED** [review]
Looks all good to me apart from the default `fh` behaviour, did you leave that in unintentionally or is this a special case where the default would make sense?

**Comment by fkiraly** [comment]
I accepted the suggestion on th `fh` default. Let's not get ahead of the resolution of our discussion in #1076.

**Comment by fkiraly** [comment]
merged since @mloning was happy except for one change request which was accepted

**Pull request merged**